modules/Emails/include/ListView/ListViewDataEmails.php

```diff
@@ -269,209 +191,817 @@ public function getListViewData(
-                            break;
-                        default:
-                            $emailRecord[strtoupper($field)] = '';
-                            break;
+                        }
                     }
-                }
 
-                $data[] = $emailRecord;
-                $pageData['rowAccess'][$h] = array('edit' => true, view => true);
-                $pageData['additionalDetails'][$h] = '';
-                $pageData['tag'][$h]['MAIN'] = 'a';
-            }
+                    return array('data'=>$data , 'pageData'=>$pageData, 'query' => $queryString);
+                    break;
+                case 'imap':
+                    // Create the data structure which are required to view a list view.
+                    require_once 'include/SearchForm/SearchForm2.php';
+                    $this->seed =& $seed;
+                    $this->setVariableName($seed->object_name, $where, $this->listviewName, $id);
+                    $this->seed->id = '[SELECT_ID_LIST]';
+
+                    if (!empty($params['overrideOrder']) && !empty($params['orderBy'])) {
+                        $order = $this->getOrderBy(
+                            strtolower($params['orderBy']),
+                            (empty($params['sortOrder']) ? '' : $params['sortOrder'])
+                        );
+                    } else {
+                        $order = $this->getOrderBy();
+                    }
+
+                    // still empty? try to use settings passed in $param
+                    if (empty($order['orderBy']) && !empty($params['orderBy'])) {
+                        $order['orderBy'] = $params['orderBy'];
+                        $order['sortOrder'] = (empty($params['sortOrder']) ? '' : $params['sortOrder']);
+                    }
 
-            $this->setVariableName($seed->object_name, $where, $this->listviewName, $id);
+                    $orderby = $order['orderBy'];
+                    if (strpos($order['orderBy'], '.') && ($order['orderBy'] != "report_cache.date_modified")) {
+                        $orderby = substr($order['orderBy'], strpos($order['orderBy'], '.') + 1);
+                    }
 
-            $this->seed->id = '[SELECT_ID_LIST]';
 
-            $nextOffset = -1;
-            $prevOffset = -1;
-            $endOffset = 0;
+                    $page = 0;
+                    $offset = 0;
+                    if (isset($_REQUEST['Emails2_EMAIL_offset'])) {
+                        if ($_REQUEST['Emails2_EMAIL_offset'] !== "end") {
+                            $offset = $_REQUEST['Emails2_EMAIL_offset'];
+                            $page = $offset / $limitPerPage;
+                        }
+                    }
 
+                    // Get emails from email server
+                    $emailServerEmails = $inboundEmail->checkWithPagination($offset, $limitPerPage, $order, $filter);
+                    // order by DESC
+                    $sortDESC = function ($a, $b) {
```

Comment:
Old code used for sorting. This was before we implemented the sort method in IMAP. 

Removed code